pkg/adapter/annotation/annotation.go

```diff
@@ -80,23 +89,36 @@ func Append(ctx context.Context, provider providerContent.Provider, desc *ocispe
 
 		return desc, nil
 
-	case ocispec.MediaTypeImageIndex:
+	// Refer: https://github.com/goharbor/harbor/blob/main/src/controller/artifact/abstractor.go#L79
+	case ocispec.MediaTypeImageIndex, images.MediaTypeDockerSchema2ManifestList:
 		var index ocispec.Index
 		labels, err = utils.ReadJSON(ctx, provider.ContentStore(), &index, *desc)
 		if err != nil {
 			return nil, errors.Wrap(err, "read manifest index")
 		}
 
+		for idx, maniDesc := range index.Manifests {
+			if maniDesc.Platform != nil && maniDesc.Platform.OSFeatures[0] == nydusutils.ManifestOSFeatureNydus {
```

Comment:
Why do we need special processing for nydus? I think appending annotations should be general.

Reply:
No, we should not add these annotations to OCI image manifest layers.